src/protocol.rs

```diff
@@ -235,8 +237,34 @@ pub fn _protocol_start(
                 });
         }));
     }
-    for thread in threads {
-        thread.join().unwrap();
+    if disable_auto_stop {
+        for thread in threads {
+            thread.join().unwrap();
+        }
+    } else {
+        tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap()
+            .block_on(async move {
+                let mut counter = 0;
+                loop {
+                    match cl.request_info().await {
+                        Ok(_) => {
+                            counter = 0;
```

Comment:
oh, I thought we only sleep when there is no error?
But this would probably also work and is more robust in some sense... at the cost of having a longer suicide waiting period. Let's just keep it as is then.